integration/samples/comments/ng-package.json

```diff
@@ -1,6 +1,7 @@
 {
   "$schema": "../../../src/ng-package.schema.json",
   "lib": {
-    "entryFile": "public_api.ts"
+    "entryFile": "public_api.ts",
+    "amdId": "@mylib/test"
```

Comment:
Hm.. Good point. Is there a way to get the name though? I am brand new to the project, so I probably just missed where you set the name..

I guess one benefit of a new setting is that it can be made optional.

Reply:
@kylecordes I like the idea of avoiding a new field.  Looks like you are using `moduleName`. My understanding is that this field is meant for something else. 

Have seen `moduleName` used for naming the module for the "global" part of the umd. (e.g ng.core, ng.forms etc). I wonder if it may cause an issue if we repurpose that field for this. I could be wrong though....

Here is how Angular is using it: https://github.com/angular/angular/blob/master/packages/core/rollup.config.js#L27

Reply:
@thelgevold  Right, I just grabbed the name at hand which seemed like it was reasonably unique and relevant to the module being bundled. If there is a way to translate the data coming from the package file and end up with a name which follows the same general convention as what Angular does, that would be great.

The particular Angular convention is pretty specific though, and ad hoc: Package is named "@angular", the translation to "ng" is clearly very angular specific. I can imagine taking the scoped NPM package name and translating it: "@foo/bar" gets an AMD module name of "foo.bar".

Somewhere while reading through the ng-packagr code, I get the impression there is something quite close to this already happening. I think the translation being done was to get something suitable to become a filename, it's possible the answer for us is as close as:

```
{ amd: { id: opts.moduleName.replace('-','.') } }
```

To get a name that is pretty close to the Angular convention, automatically?

Reply:
I am thinking now that it may be a good idea to keep them separate to mirror the Rollup api.
There are scenarios where they are pretty different as well.
Example: `@angular/forms` vs `ng.form`. 

At least `moduleName` has to be a valid JavaScript property name since it may be used as a global variable if you load the "global" part of the umd. 

Example `gobal.ng.form` for someone who loads the umd as a script tag and not AMD or CommonJS.

Reply:
I think it's obviously helpful to be able to support something like the Angular case where there is a different name applied for various historical or ad hoc reasons.

However, unless it is made to default to something based on the package name, it is likely that most packages published using `ng-packagr` will (continue to) not have an AMD module ID, they will be anonymous...  which will not help in nudging packages toward shipping things that work with the Bazel `ts_devserver` that @alexeagle pointed out as the reason for his issue.

Should ng-packagr make it *possible* to make something that works with devserver, or make it the *default* for the output to work with the devserver?

Reply:
I am in favour of @Toxicable's proposal: make a reasonable default [based on the module id](https://github.com/dherges/ng-packagr/blob/master/src/lib/ng-package-format/entry-point.ts#L110-L121), make it optional to override with a option.